packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -845,8 +845,43 @@ export const getBlockSelectors = ( blockTypes ) => {
 	return result;
 };
 
+/**
+ * If there is a separator block whose color is defined in theme.json via background,
+ * update the separator color to the same value by using border color.
+ *
+ * @param {Object} config Theme.json configuration file object.
+ * @return {Object} configTheme.json configuration file object updated.
+ */
+function updateConfigWithSeparator( config ) {
+	const needsSeparatorStyleUpdate =
+		config.styles?.blocks[ 'core/separator' ] &&
+		config.styles?.blocks[ 'core/separator' ].color?.background &&
+		! config.styles?.blocks[ 'core/separator' ].color?.text &&
+		! config.styles?.blocks[ 'core/separator' ].border?.color;
+	if ( needsSeparatorStyleUpdate ) {
+		return {
+			...config,
+			styles: {
+				...config.styles,
+				blocks: {
+					...config.styles.blocks,
+					'core/separator': {
+						...config.styles.blocks[ 'core/separator' ],
+						color: {
+							text: config.styles?.blocks[ 'core/separator' ]
```

Comment:
Apologies for missing this earlier, I just noticed that this object is missing a spread before `text` ( e.g. `...config.styles.blocks[ 'core/separator' ].color`, so it's currently overwriting the `background` prop. In global styles in the site editor, this means that `background` is not being applied to the custom block styles that require it:

<img width="854" alt="image" src="https://user-images.githubusercontent.com/14988353/197688955-ce966024-1b38-4446-8f7e-d0a12bd681e6.png">

In the screenshot above, the `text` color is correctly using the green, but the diamond style's default black is not being overridden because the `background-color` key isn't being output.

I've pushed a commit in https://github.com/WordPress/gutenberg/pull/44943/commits/ed4a033786718597f9434663798e18ccc17b566b to fix this up.